aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -816,19 +816,19 @@ static void add_out_dense_sparse_compressed_cpu(
   TORCH_INTERNAL_ASSERT(dense.layout() == kStrided);
   TORCH_INTERNAL_ASSERT(
       src.layout() == kSparseCsr || src.layout() == kSparseCsc);
-  TORCH_INTERNAL_ASSERT(dense.device() == kCPU);
+  TORCH_INTERNAL_ASSERT(dense.device() == kCPU || dense.device() == kMeta);
```

Comment:
Is this actually OK though, I see below we start doing pointer accessors...

Reply:
There is a bail-out on nnz == 0 before that, taken for meta. I can keep the kCPU assert after that if you prefer that certainty!

Reply:
We did nnz = 0 for meta because it was an easy way to hack things up to work, but you can imagine a universe where we support non-zero nnz on meta tensor...

So yes please.